src/preferences/configobject.h

```diff
@@ -177,6 +177,8 @@ template <class ValueType> class ConfigObject {
     void reopen(const QString& file);
     bool save();
 
+    static QString computeResourcePath();
```

Comment:
Why not return a `QDir` which would simplify composing paths?

Reply:
We do not need QDir facilities here. A QDir constructor allocates a new QDirPrivate object with some unneeded members, including a deep copy of the given path.

Reply:
Concise code >> optimized code.

Reply:
I do not consider this as unsafe. QDir is written to handle native path separators. It assumes native path separators in all input functions, which is not the case here. 
See: https://github.com/qt/qtbase/blob/9db7cc79a26ced4997277b5c206ca15949133240/src/corelib/io/qdir.cpp#L190
All other Qt functions take paths as QString once we are in the Qt separator domain.  So it should be safe to follow this pattern.

Reply:
> The QDir is supposed to be created only once

There is not much of caching here, since every test instantiates a new test class as base class. executed by ctest it is even an independent process.

Reply:
I agree that we should favor a QDir over a QString. Otherwise we end up with a "stringly-typed system" (which we already have enough of in the form of COs).